pkg/process/metadata/parser/service_test.go

```diff
@@ -213,6 +213,27 @@ func TestExtractServiceMetadata(t *testing.T) {
 			},
 			expectedServiceTag: "process_context:myservice-core",
 		},
+		{
+			name: "dotnet cmd with dll",
+			cmdline: []string{
+				"/usr/bin/dotnet", "./myservice.dll",
+			},
+			expectedServiceTag: "process_context:myservice",
+		},
+		{
+			name: "dotnet cmd with dll and options",
+			cmdline: []string{
+				"/usr/bin/dotnet", "-v", "--", "/app/lib/myservice.dll",
+			},
+			expectedServiceTag: "process_context:myservice",
+		},
+		{
+			name: "dotnet cmd with unrecognized options",
+			cmdline: []string{
+				"/usr/bin/dotnet", "run", "--project", "./projects/proj1/proj1.csproj",
+			},
+			expectedServiceTag: "process_context:dotnet",
```

Comment:
Do we do this for other services as well? (like java)

Reply:
yes the same apply for java (`java` is the safe default). There is another PR for node that uses `node` as default. Python seems differing since returning empty string. Perhaps this should be harmonized but in case it will be addressed in another PR